### DISCUSSION TITLE
[mini.comment] Does not uncomment C's /* syntax correctly

### DISCUSSION THREAD

**Discussion by ldelossa** [nvim-mini/mini.nvim#1382] [Q&A] [answered]
Hey there.

Wondering if this is a fixable issue with mini.comments. 

I work on a C codebase which uses the '/* */' comment syntax. 

It appears that `mini.comment` can create comments correctly, but once the comment is inserted, it doesn't understand how to uncomment it, and instead adds another comment in the `//` comment syntax. 

I've uploaded a screencap to explain. 

I read thru the help but nothing stood out to me on how to fix this. 

https://github.com/user-attachments/assets/5784744d-94cd-4666-b74d-b137758a24ad

**Comment by echasnovski** [comment, marked as answer]
'mini.comment' [uses 'commentstring' option](https://github.com/echasnovski/mini.nvim/blob/690a3b4c78c4956f7ecf770124b522d32084b872/doc/mini-comment.txt#L11-L14) to determine comment structure. It is taken either from local tree-sitter language (useful in, for example, markdown code blocks) or buffer's option value. There is also a [`config.options.custom_commentstring`](https://github.com/echasnovski/mini.nvim/blob/690a3b4c78c4956f7ecf770124b522d32084b872/doc/mini-comment.txt#L120) to programmatically override in some complex cases (like JSX files).

That said, I can not reproduce this. So:

- What Neovim version are you using?
- What does executing `:=vim.bo.commentstring` (buffer-local 'commentstring' value) show?
- Do you have `config.option.custom_commentstring` defined? You can check it with `:=MiniComment.config.option.custom_commentstring` (should be `nil` if nothing is defined).
- Do you have 'comment' tree-sitter parser installed? If you are using 'nvim-treesitter/nvim-treesitter', you can check it with `:TSInstallInfo`.

**Reply by ldelossa** [reply]
Thank you @echasnovski 

Your hint about `commentstring` helped me find the issue. 

I set comment string depending on particular repositories. I had an autocommand which must have been conflicting with mini.comment. I removed this and all is well. 

Thanks again.
